ca/transport.go

```diff
@@ -44,6 +44,23 @@ func (c *MutableTLSCreds) Info() credentials.ProtocolInfo {
 	}
 }
 
+// Clone returns new MutableTLSCreds created from underlying *tls.Config.
+// It returns nil if validation of underlying config fails.
+func (c *MutableTLSCreds) Clone() credentials.TransportCredentials {
+	c.Lock()
+	newCfg, _ := NewMutableTLS(c.config)
```

Comment:
Since `c.config` is a pointer - are we actually cloning? Does this interface expect a deep copy?